### DISCUSSION TITLE
Can I only import the exact code I used in the project ?

### DISCUSSION THREAD

**Discussion by hardylab** [dexie/Dexie.js#1524] [Q&A]
Since many of the Dexie's API are not in use, to minify the js bundles, I hope to only import the exact code I need to use in my project, how to do this? I are not sure which code is really standalone.

**Comment by dfahlander** [comment]
Only explicit exports can be tree-shaken away. There aren't so much in dexie that can be tree-shaken expect `liveQuery`, `mergeRanges` and `rangesOverlap`. So if you don't import any of these, you will get a slightly lighter bundle than if they are imported (assuming you are using a bundler such as webpack or rollup).
